### PR TITLE
Feature/semantic scrub

### PR DESCRIPTION
**Pull request by KrishPatel13**
**What kind of change does this PR introduce?**
Draft PR for Semantic Scrubbing:

Part 1:
A:
- [x] Semantic Scrubbing without specific identification.
- [x] Test Semantic Scrubbing with visualization.

B:
- [ ] Specific Entity Identification. as shown in the eg. in the Issue )

**Checklist**
* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] If applicable, I have added tests to prove my fix is functional/effective
* [x] I have linted my code locally prior to submission
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation (e.g. README.md, requirements.txt)
* [x] New and existing unit tests pass locally with my changes

**Summary**
Progress on Semantic Scrubbing test.

Motivation: [Issue @264](https://github.com/MLDSAI/OpenAdapt/issues/264)

**How can your code be run and tested?**

**Other information**

### CODE REVIEW

**Comment by KrishPatel13** [comment]
The current version of the PR, replaces the identified PII/PHI with this entity types.

Before:
```
LLC OPERATING AGREEMENT
This LLC Operating Agreement (“Agreement”) is entered into on )2/06/22	Effective Date”). The members in this Agreement are as follows:

Member Name:	John  Doe		  Address:	125 Military Drive, Washington DC, 20001 Member Name:	Tom Jerry	 Address:	25 Empire State, New York, 20002		

Above individuals are referred to herein individually as “Member” and together as “Members”.

Recitals

 
The company will be known as ABC LLC	
of 16/06/2023	
 
(“Company”) with a formation date
 

The Company’s primary place of business will be Sans Fransisco	       The Company will be governed under the laws of the state of California	
The Company’s primary purpose is to provide Legal Support	

Agreement

NOW, THEREFORE, in consideration of the respective covenants and agreements contained in this Agreement, Members agree as follows:

1.	INTEREST AND AUTHORITY
a.	The Members’ ownership interest in the Company will be as follows:

John	Class A
 	 

Tom	Class B


b.	All Members will have equal vote. No Member is authorized to act on their own in obtaining contracts, financial, or other obligations on the Company. Decisions will be based on a majority of equal votes.
 
2.	CAPITAL CONTRIBUTIONS
a.	Members shall make a capital contribution in such amounts as listed below. The Members may take distributions of the capital from time to time in accordance with the limitations imposed by this Agreement.

 	JOhn : capital contribution of	15000	

 	Tom : capital contribution of	200000	

3.	PROFIT AND LOSSES
a.	The Members will share the net profits and net loss according to the following percentages:
John	50
Tom	50
```


After:
```
LLC OPERATING AGREEMENT
This LLC Operating Agreement (“Agreement”) is entered into on )2/06/22	Effective Date”). The members in this Agreement are as follows:

Member Name:	<PERSON>		  Address:	125 <LOCATION>, <LOCATION>, 20001 Member Name:	<PERSON>	 Address:	25 <LOCATION>, <LOCATION>, 20002		

Above individuals are referred to herein individually as “Member” and together as “Members”.

Recitals

 
The company will be known as ABC LLC	
of 16/06/2023	
 
(“Company”) with a formation date
 

The Company’s primary place of business will be Sans Fransisco	       The Company will be governed under the laws of the state of <LOCATION>	
The Company’s primary purpose is to provide Legal Support	

Agreement

NOW, THEREFORE, in consideration of the respective covenants and agreements contained in this Agreement, Members agree as follows:

1.	INTEREST AND AUTHORITY
a.	The Members’ ownership interest in the Company will be as follows:

John	Class A
 	 

Tom	Class B


b.	All Members will have equal vote. No Member is authorized to act on their own in obtaining contracts, financial, or other obligations on the Company. Decisions will be based on a majority of equal votes.
 
2.	CAPITAL CONTRIBUTIONS
a.	Members shall make a capital contribution in such amounts as listed below. The Members may take distributions of the capital from time to time in accordance with the limitations imposed by this Agreement.

 	<PERSON> : capital contribution of	15000	

 	<PERSON> : capital contribution of	<US_DRIVER_LICENSE>	

3.	PROFIT AND LOSSES
a.	The Members will share the net profits and net loss according to the following percentages:
<PERSON>	50
<PERSON>	50
```

**Comment by KrishPatel13** [comment]
Checked Task 1 as completed

**Comment by KrishPatel13** [comment]
Step 2:  Testing with Visualization

Some Screenshots to demonstrate how Visualization looks like with semantics:

image

**Comment by KrishPatel13** [comment]
image

**Comment by KrishPatel13** [comment]
image

**Comment by KrishPatel13** [comment]
Marking Step 2 as Completed.

Some Notes:
> If we have a name, then it scrub to <PERSON>.
> And during visualization, We would see it as <-P-E-R-S-O-N-> in the textbox in the diff images as well as on the right in the text. Also seen in the above screenshots.

**Comment by KrishPatel13** [comment]
After the above visualization fix, 
the redaction is Blue in color despite the SCRUB_FILL_COLOR - 0xFF0000 which is Red (in hex).

**Comment by abrichr** [comment]
Thanks @KrishPatel13 ! As discussed let's please ignore visualization for now and focus on:

1. Scrubbing entities (a: default entities, e.g. firstname/lastname; b: LLC names)
2. Retaining semantic information

It seems Step 1a is complete. Let's focus on Step 1b using lex-nlp.

**Comment by KrishPatel13** [comment]
@abrichr Part 1a should be ready to be merged! This is Does NOT completely resolve https://github.com/MLDSAI/OpenAdapt/issues/264

So I request you keep it open but only merge this. 🙏

File: tests/openadapt/test_scrub.py [hunk 1]
```diff
@@ -9,6 +9,18 @@
 from openadapt import scrub, config
 
 
+def _hex_to_rgb(hex_color: int) -> tuple[int, int, int]:
+    """
+    Convert a hex color (int) to RGB
+    """
+
+    assert 0x000000 <= hex_color <= 0xFFFFFF
```
Comment (abrichr) [inline tests/openadapt/test_scrub.py:17]: ❤️

File: tests/openadapt/test_scrub.py [hunk 2]
```diff
@@ -9,6 +9,18 @@
...
+    b = (hex_color >> 16) & 0xFF
+    g = (hex_color >> 8) & 0xFF
+    r = hex_color & 0xFF
```
Comment (abrichr) [inline tests/openadapt/test_scrub.py:20]: 🥷

**Comment by abrichr** [comment]
🙏 🚀

**Review by abrichr — APPROVED** [review]
🙏 🚀

**Comment by KrishPatel13** [comment]
Rename the title to `Pseudonymization`
In Favour of: https://medium.com/@openredact/anonymizer-a-framework-for-text-anonymization-499855f639d4

**Pull request merged**